### PR TITLE
[rls/daal-2020-mnt] Finiteness checker

### PR DESCRIPTION
**Pull request by Alexsandruss**
NaN/Inf checker for Numeric Tables with base and AVX512 branches for float32/64 types.
Checker has two stages:

1. Compute sum of all values and check its finiteness
2. If sum isn't finite, check finiteness of all values.

### CODE REVIEW

**Comment by Alexsandruss** [comment]
http://intel-ci.intel.com/ea4daacb-c29f-f1a6-9870-8cdcd4b723a1

**Comment by Alexsandruss** [comment]
http://intel-ci.intel.com/ea4dac4e-0ed5-f1da-b9d2-8cdcd4b723a1

File: service/kernel/data_management/finiteness_checker.cpp [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+{
+namespace data_management
+{
+namespace internal
+{
+typedef daal::data_management::NumericTable::StorageLayout NTLayout;
+
+void valuesAreFinite(const float * dataPtr, bool * finitenessPrt, size_t n, bool allowNaN)
+{
+    uint32_t * uint32Ptr = (uint32_t *)dataPtr;
```
Comment (ShvetsKS) [inline service/kernel/data_management/finiteness_checker.cpp]: const uint32_t * const
And similar for all code lines below, follow coding style.
sorry for not covering it in previous review.

File: service/kernel/data_management/finiteness_checker.cpp [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+template <>
+bool checkFiniteness<double, avx512>(size_t nElements, size_t nDataPtrs, size_t nElementsPerPtr, size_t optArrLen, const double ** dataPtrs, bool * finiteness, bool allowNaN)
+{
+    return checkFinitenessAVX512Impl<double>(nElements, nDataPtrs, nElementsPerPtr, optArrLen, dataPtrs, finiteness, allowNaN);
+}
+
+#endif
+
+template <typename DataType, daal::CpuType cpu>
+services::Status allValuesAreFiniteImpl(NumericTable & table, bool allowNaN, bool * finiteness)
```
Comment (ShvetsKS) [inline service/kernel/data_management/finiteness_checker.cpp]: not applied comment:
const NumericTable&

File: service/kernel/data_management/finiteness_checker.cpp [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+{
+typedef daal::data_management::NumericTable::StorageLayout NTLayout;
+
+void valuesAreFinite(const float * dataPtr, bool * finitenessPrt, size_t n, bool allowNaN)
+{
+    uint32_t * uint32Ptr = (uint32_t *)dataPtr;
+    for (size_t i = 0; i < n; ++i)
+    {
+        // first check: all exponent bits are 1 (inf or nan)
+        if (0x7f800000u == (uint32Ptr[i] & 0x7f800000u))
```
Comment (ShvetsKS) [inline service/kernel/data_management/finiteness_checker.cpp]: create const variables or defines for numbers like `0x7f800000u`
applicable for below

File: service/kernel/data_management/finiteness_checker.cpp (hunk 1 above)
Comment (Alexsandruss) [reply service/kernel/data_management/finiteness_checker.cpp]: Corrected

File: service/kernel/data_management/finiteness_checker.cpp (hunk 3 above)
Comment (Alexsandruss) [reply service/kernel/data_management/finiteness_checker.cpp]: Constants are created.

**Comment by Alexsandruss** [comment]
http://intel-ci.intel.com/ea50d362-52da-f1a1-ba17-8cdcd4b723a1

**Pull request merged**